server/src/services/initialization/onInitialize.ts

```diff
@@ -40,22 +44,49 @@ export const onInitialize = (serverState: ServerState) => {
       workspaceFolders,
     });
 
-    // fetch available solidity versions
-    await updateAvailableSolcVersions(serverState);
+    // fetch available solidity versions and feature flags
+    const [flags] = await Promise.all([
+      fetchFeatureFlags(serverState),
+      updateAvailableSolcVersions(serverState),
+    ]);
 
     logger.info("Language server ready");
 
+    const slangSupported = isSlangSupported();
+
+    const semanticTokensEnabled = isFeatureEnabled(
+      serverState,
+      flags,
+      "semanticHighlighting",
+      machineId
+    );
+
+    const documentSymbolsEnabled = isFeatureEnabled(
+      serverState,
+      flags,
+      "documentSymbol",
+      machineId
+    );
     // Index and analysis
-    await serverState.telemetry.trackTiming("indexing", async (transaction) => {
-      await indexWorkspaceFolders(
-        serverState,
-        serverState.workspaceFileRetriever,
-        workspaceFolders,
-        transaction
-      );
-
-      return { status: "ok", result: null };
-    });
+    await serverState.telemetry.trackTiming(
+      "indexing",
+      async (transaction) => {
+        await indexWorkspaceFolders(
+          serverState,
+          serverState.workspaceFileRetriever,
+          workspaceFolders,
+          transaction
+        );
+
+        return { status: "ok", result: null };
+      },
+      {
```

Comment:
I wonder if we should log these metadata as part of the `indexing` event, or a slamg-specific one, as it is unrelated to indexing the workspace.

Reply:
I see. Thanks for explaining.